app/models/middleware_server_performance.rb

```diff
@@ -16,7 +16,11 @@ def self.build_results_for_report_middleware(options)
     start_time, end_time, interval = parse_time_interval(options)
     results = []
     MiddlewareServer.find_each do |ms|
-      raw_stats = fetch_raw_stats(ms, start_time, end_time, interval)
+      begin
+        raw_stats = fetch_raw_stats(ms, start_time, end_time, interval)
```

Comment:
This is my initial test. As you see there are a lot of exceptions, IIRC the base of each of them is StandardError, so each need to be handled independtly. I was hoping to have only one (base)exception to handle in case of a connection exception and avoid to have this kind of handling on MiQ code.